test/mitmproxy/test_command.py

```diff
@@ -309,6 +309,29 @@ def empty(self) -> None:
         pass
 
 
+class TAttr:
+    def __getattr__(self, item):
+        raise IOError
+
+
+class TCmds(TAttr):
+    def __init__(self):
+        self.TAttr = TAttr()
+
+
+def test_collect_commands():
+    """
+        This tests for the error thrown by hasattr()
+    """
+    with taddons.context() as tctx:
+        c = command.CommandManager(tctx.master)
+        a = TCmds()
+        c.collect_commands(a)
+        assert "cmd1" not in c.commands
+        assert "cmd2" not in c.commands
+        assert "empty" not in c.commands
```

Comment:
Much better! The asserts don't make sense though, maybe we can add an actual command that still should be loaded?